addons/addon-base-raas/packages/base-raas-services/lib/plugins/env-provisioning-plugin.js

```diff
@@ -212,11 +216,15 @@ async function updateEnvOnTerminationSuccess({ requestContext, container, status
   log.debug({ msg: `Cleaning up local resource policies`, envId });
 
   // Delete DNS record for RStudio workspaces
-  const connectionType = _.find(updatedEnvironment.outputs, x => x.OutputKey === 'MetaConnection1Type').OutputValue;
-  if (connectionType.toLowerCase() === 'rstudio') {
-    const dnsName = _.find(updatedEnvironment.outputs, x => x.OutputKey === 'Ec2WorkspaceDnsName').OutputValue;
-    const environmentDnsService = await container.find('environmentDnsService');
-    environmentDnsService.deleteRecord('rstudio', envId, dnsName);
+  const connectionType = _.find(updatedEnvironment.outputs, o => o.OutputKey === 'MetaConnection1Type');
+  let connectionTypeValue;
+  if (connectionType) {
+    connectionTypeValue = connectionType.OutputValue;
```

Comment:
Can we add a couple of unit tests for this branch for both create and delete?

Reply:
Per our conversation, the underlying service would see some changes during the auto-stop feature addition. Implementing unit tests then.